octue/cloud/pub_sub/service.py

```diff
@@ -271,12 +254,19 @@ def ask(
         return response_subscription, question_uuid
 
     def wait_for_answer(
-        self, subscription, service_name="REMOTE", timeout=60, delivery_acknowledgement_timeout=30, retry_interval=5
+        self,
+        subscription,
+        monitoring_callback=None,
```

Comment:
Should be `handle_monitor_message` - note also discussion to update the strand name to `monitor_message_schema`

- `monitor` rather than `monitoring` because it's a noun
- `monitor_message_schema` rather than `monitor_messages_schema` because this is a schema to handle a single message not multiple ones. Note that different ***formats*** of message can be absorbed using `anyOf` within the schema